src/search_renderer.cpp

```diff
@@ -93,30 +93,38 @@ std::string SearchRenderer::getHtml()
   kainjow::mustache::data pages{kainjow::mustache::data::type::list};
 
   auto resultStart = mp_searcher->getResultStart();
-  auto resultEnd = mp_searcher->getResultEnd();
-  auto resultCountPerPage = resultEnd - resultStart;
+  auto pageLength = this->pageLength;
   auto estimatedResultCount = mp_searcher->getEstimatedResultCount();
   auto currentPage = 0U;
   auto pageStart = 0U;
   auto pageEnd = 0U;
   auto lastPageStart = 0U;
-  if (resultCountPerPage) {
-    currentPage = resultStart/resultCountPerPage;
+  if (pageLength) {
+    currentPage = resultStart/pageLength;
     pageStart = currentPage > 4 ? currentPage-4 : 0;
     pageEnd = currentPage + 5;
-    if (pageEnd > estimatedResultCount / resultCountPerPage) {
-      pageEnd = estimatedResultCount / resultCountPerPage;
+    if (pageEnd > estimatedResultCount / pageLength) {
+      pageEnd = estimatedResultCount / pageLength;
+      //to include last page in pagination if page is not empty
+      if(pageEnd*pageLength!=estimatedResultCount)
+      {
+        pageEnd++;
+      }
     }
-    if (estimatedResultCount > resultCountPerPage) {
-      lastPageStart = static_cast<int>(round(estimatedResultCount/resultCountPerPage)) * resultCountPerPage;
+    if (estimatedResultCount > pageLength) {
+      lastPageStart = static_cast<int>(round(estimatedResultCount/pageLength)) * pageLength;
+      //changing last page if it is not empty
+      if(lastPageStart==estimatedResultCount && lastPageStart>pageLength)
+      {
+        lastPageStart-=pageLength;
+      }
```

Comment:
I think this can be simplified to

```
lastPageStart = ((estimatedResultCount-1)/pageLength)) * pageLength;
```

since `(estimatedResultCount-1)/pageLength` is the index of the page containing the last result.